soh/src/overlays/actors/ovl_Item_B_Heart/z_item_b_heart.c

```diff
@@ -99,8 +99,8 @@ void ItemBHeart_Draw(Actor* thisx, PlayState* play) {
     }
 
     if (IS_RANDO) {
-        GetItemEntry_Draw(play, Randomizer_GetItemFromActor(this->actor.id, 
-            play->sceneNum,this->actor.params, GI_HEART_CONTAINER_2));
+        RandomizerCheck check = Randomizer_GetCheckFromActor(this->actor.id, play->sceneNum, this->actor.params);
+        GetItemEntry_Draw(play, (CVarGetInteger("gRandoEnhancements.MysteriousShuffle", 0) && Randomizer_IsCheckShuffled(check)) ? GetItemMystery : Randomizer_GetItemFromKnownCheck(check, GI_HEART_CONTAINER_2), check);
```

Comment:
removing the extra `check` at the end gets past the error that is visible in CI at the moment, but when trying to build locally after that change i started getting a ton of multiple definition errors (the "move to a function" thing makes those go away)

```suggestion
        GetItemEntry_Draw(play, (CVarGetInteger("gRandoEnhancements.MysteriousShuffle", 0) && Randomizer_IsCheckShuffled(check)) ? GetItemMystery() : Randomizer_GetItemFromKnownCheck(check, GI_HEART_CONTAINER_2));
```